python/ray/tests/test_global_state.py

```diff
@@ -154,6 +155,39 @@ def ready(self):
     )
 
 
+def test_node_table(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(node_name="head_node", include_dashboard=False)
+    head_context = ray.init(address=cluster.address, include_dashboard=False)
+    cluster.add_node(node_name="worker_node", include_dashboard=False)
+    cluster.wait_for_nodes()
+
+    global_state_accessor = make_global_state_accessor(head_context)
+    node_table = global_state_accessor.get_node_table()
+    assert len(node_table) == 2
+    for node_data in node_table:
+        node = gcs_utils.GcsNodeInfo.FromString(node_data)
+        if (
+            ray._private.utils.binary_to_hex(node.node_id)
+            == head_context.address_info["node_id"]
+        ):
+            assert node.node_name == "head_node"
+        else:
+            assert node.node_name == "worker_node"
+
+    global_state_accessor.disconnect()
+    ray.shutdown()
+    cluster.shutdown()
+
+    # Test ray.init with _node_name directly
+    new_head_context = ray.init(_node_name="new_head_node", include_dashboard=False)
+
+    global_state_accessor = make_global_state_accessor(new_head_context)
+    node_data = global_state_accessor.get_node_table()[0]
+    node = gcs_utils.GcsNodeInfo.FromString(node_data)
+    assert node.node_name == "new_head_node"
```

Comment:
Can you also add a test when the node is dead by heartbeat missing, the information contains the node name?